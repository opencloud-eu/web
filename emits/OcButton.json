[
  {
    "name": "click",
    "description": "Emitted when the button has been clicked.",
    "type": "(e: 'click', event: MouseEvent): void"
  }
]