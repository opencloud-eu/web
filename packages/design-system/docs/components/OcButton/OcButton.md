---
title: OcButton component
next: false
prev: false
---

# OcButton component

## Description

The `OcButton` component is supposed to be used for interface actions. It's suitable for all-purpose use.

It defaults to an outlined button with a white background. A filled appearance should be used only once per view for the main call-to-action. All buttons are built with a css grid which ensures that there will be a pre-defined gutter between all child items.

## Accessibility

### Distinction when to use `<button>` and when to use an `<a>`

Regardless of the visual representation the following differentiation should be made if in doubt what to choose:

- An anchor/link changes the location, be it internally on the website, or to another resource/document/route.
- A button does change the state of the website, e.g.: adds, deletes, opens, ...

### Accessible name

The `accessible-name` of a button is derived from several sources. Right now, only two of them are relevant:

- The value of the `aria-label` attribute
- The text between the opening and closing tag: `<button>This text here</button>`

When an `aria-label` attribute exists, its value will override the button text. So in this case, the accessible name would be "foo": `<button aria-label="foo">Bar</button>`, although visual users will see "Bar". This difference between accessible name and visual name is a problem for a certain type of assistive technology (explainer for the term), this is why the WCAG success criterion 2.5.3, "Label in name" exists. This difference should be avoided, if it can't, W3C recommends that the accessible name should start with visible label.

### Icon-only button

Every icon-only button has to have an `aria-label` that describes the purpose of the button.

## Examples

### Appearance

::: livecode

```html
<oc-button appearance="filled">Filled</oc-button>
<oc-button appearance="outline">Outline</oc-button>
<oc-button appearance="raw" class="oc-p-s oc-ml-xs">Raw</oc-button>
<div class="oc-p-s oc-mt-m" style="background: #000000;">
  <oc-button appearance="raw-inverse" class="oc-p-s">Raw-inverse</oc-button>
</div>
```

:::

### Icons

::: livecode

```html
<oc-button appearance="filled"><oc-icon name="home" /><span>Home</span></oc-button>
<oc-button><oc-icon name="home" /><span>Home</span></oc-button>
<oc-button aria-label="Go to your home"><oc-icon name="home" /></oc-button>
<oc-button aria-label="Go to your home" appearance="raw" class="oc-p-s oc-ml-xs"
  ><oc-icon name="home"
/></oc-button>
```

:::

### Groups

::: livecode

```html
<div class="oc-button-group">
  <oc-button>Foo</oc-button>
  <oc-button>Bar</oc-button>
  <oc-button appearance="filled">Baz</oc-button>
</div>
```

:::

### Click handler

A click handler can be registered via the `@click` property.

::: livecode {path=/components/OcButton/handler.vue}
<<< @/components/OcButton/handler.vue
:::

::: component-api
