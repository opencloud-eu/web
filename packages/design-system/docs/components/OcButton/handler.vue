<template>
  <oc-button @click="increaseCounter">Increase counter</oc-button>
  <oc-button class="ml-4" @click="resetCounter">Reset counter</oc-button>
  <div class="mt-4">Current count: {{ counter }}</div>
</template>

<script setup lang="ts">
import { ref } from 'vue'

const counter = ref(0)

const increaseCounter = () => {
  counter.value++
}

const resetCounter = () => {
  counter.value = 0
}
</script>
