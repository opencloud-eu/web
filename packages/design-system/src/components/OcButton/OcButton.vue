<template>
  <component
    :is="type"
    v-bind="additionalAttributes"
    :aria-label="ariaLabel"
    :class="buttonClass"
    v-on="handlers"
  >
    <oc-spinner v-if="showSpinner" size="small" class="spinner" />
    <!-- @slot Content of the button -->
    <slot />
  </component>
</template>

<script setup lang="ts">
import { computed } from 'vue'
import { RouteLocationRaw } from 'vue-router'
import { AppearanceType, getSizeClass, SizeType } from '../../helpers'
import { kebabCase } from 'lodash-es'

export interface Props {
  /**
   * @docs The appearance of the button.
   * @default outline
   */
  appearance?: AppearanceType
  /**
   * @docs The aria label of the button. Needs to be present if the button doesn't have a visible label.
   */
  ariaLabel?: string
  /**
   * @docs Material design color role.
   * @default secondary
   */
  colorRole?:
    | 'primary'
    | 'primaryContainer'
    | 'primaryFixed'
    | 'secondary'
    | 'secondaryContainer'
    | 'secondaryFixed'
    | 'tertiary'
    | 'tertiaryContainer'
    | 'tertiaryFixed'
    | 'surface'
    | 'surfaceContainer'
    | 'chrome'
  /**
   * @docs Determines if the button is disabled.
   * @default false
   */
  disabled?: boolean
  /**
   * @docs The gap size between content elements of the button.
   * @default medium
   */
  gapSize?: SizeType | 'none'
  /**
   * @docs The href if the `type` is set to `a'.
   */
  href?: string
  /**
   * @docs The alignment of the button content.
   * @default center
   */
  justifyContent?: 'left' | 'center' | 'right' | 'space-around' | 'space-between' | 'space-evenly'
  /**
   * @docs Determines if a spinner should be shown inside the button.
   * @default false
   */
  showSpinner?: boolean
  /**
   * @docs The size of the button.
   * @default medium
   */
  size?: 'small' | 'medium' | 'large'
  /**
   * @docs The type of the button element. Only takes effect if the `type` is set to `button`.
   * @default button
   */
  submit?: 'null' | 'button' | 'submit' | 'reset'
  /**
   * @docs The target of the button if the `type` is set to `a`.
   */
  target?: '_blank' | '_self' | '_parent' | '_top'
  /**
   * @docs The route location if the `type` is set to `router-link`.
   */
  to?: RouteLocationRaw
  /**
   * @docs The type of the button element.
   * @default button
   */
  type?: 'button' | 'a' | 'router-link'
  /**
   * @docs Determines if the button should have no hover effect.
   * @default false
   */
  noHover?: boolean
}

export interface Emits {
  /**
   * @docs Emitted when the button has been clicked.
   */
  (e: 'click', event: MouseEvent): void
}

export interface Slots {
  /**
   * @docs Button content.
   */
  default?: () => unknown
}

const {
  appearance = 'outline',
  ariaLabel,
  colorRole = 'secondary',
  disabled = false,
  gapSize = 'medium',
  href,
  justifyContent = 'center',
  showSpinner = false,
  size = 'medium',
  submit = 'button',
  target,
  to,
  type = 'button',
  noHover = false
} = defineProps<Props>()

const emit = defineEmits<Emits>()
defineSlots<Slots>()

const buttonClass = computed(() => {
  const classes = [
    'oc-button',
    'oc-rounded',
    `oc-button-${getSizeClass(size)}`,
    `oc-button-justify-content-${justifyContent}`,
    `oc-button-gap-${getSizeClass(gapSize)}`,
    `oc-button-${kebabCase(colorRole)}`,
    `oc-button-${appearance}`,
    `oc-button-${kebabCase(colorRole)}-${appearance}`
  ]
  if (noHover) {
    classes.push('no-hover')
  }
  return classes
})

const additionalAttributes = computed(() => {
  return {
    ...(href && { href }),
    ...(target && { target }),
    ...(to && { to }),
    ...(type === 'button' && { type: submit }),
    ...(type === 'button' && { disabled })
  }
})

const handlers = computed(() => {
  return {
    ...(type === 'button' && { click: onClick })
  }
})

const onClick = (event: MouseEvent) => {
  emit('click', event)
}
</script>

<style lang="scss">
@mixin oc-button-gap($factor) {
  gap: math.round(calc($oc-space-small * $factor / 2)) * 2;
}

@mixin oc-button-line-height($factor) {
  line-height: $oc-size-icon-default * $factor;
}

@mixin oc-button-color-role($color, $on-color) {
  &-raw,
  &-raw-inverse {
    border-style: none;
    font-size: var(--oc-font-size-medium);
    font-weight: normal;
    min-height: 0;
    padding: 0;

    background-color: transparent;
    color: $color;
    .oc-icon > svg {
      fill: $color;
    }

    &:focus:not([disabled]):not(button),
    &:hover:not([disabled]):not(button) {
      background-color: transparent;
    }

    &:focus:not([disabled]):not(.active):not(.no-hover),
    &:hover:not([disabled]):not(.active):not(.no-hover) {
      background-color: var(--oc-role-surface-container);
      color: var(--oc-role-on-surface);
      .oc-icon > svg {
        fill: var(--oc-role-on-surface);
      }
    }
  }
  &-raw-inverse {
    color: $on-color;
    .oc-icon > svg {
      fill: $on-color;
    }
  }

  &-filled {
    background-color: $color;
    color: $on-color;
    .oc-icon > svg {
      fill: $on-color;
    }
  }

  &-outline {
    outline: 1px solid $color;
    outline-offset: -1px;
    background-color: transparent;
    color: $color;
    .oc-icon > svg {
      fill: $color;
    }
  }
}

.oc-button {
  @include oc-button-line-height(1);

  align-items: center;
  border: 0;
  box-sizing: border-box;
  display: inline-flex;
  font-weight: 400;
  padding: 0.5rem 0.8rem;
  text-align: left;
  text-decoration: none;

  &-justify-content {
    &-left {
      justify-content: left;
    }

    &-center {
      justify-content: center;
    }

    &-right {
      justify-content: right;
    }

    &-space-between {
      justify-content: space-between;
    }

    &-space-around {
      justify-content: space-around;
    }

    &-space-evenly {
      justify-content: space-evenly;
    }
  }

  &-gap {
    &-xs {
      @include oc-button-gap(0.5);
    }

    &-s {
      @include oc-button-gap(0.7);
    }

    &-m {
      @include oc-button-gap(1);
    }

    &-l {
      @include oc-button-gap(1.5);
    }

    &-xl {
      @include oc-button-gap(2);
    }
  }

  &:hover {
    cursor: pointer;
  }

  &-s {
    @include oc-button-line-height(0.7);

    font-size: var(--oc-font-size-small);
    min-height: 1.2rem;
  }

  &-m {
    @include oc-button-line-height(1);

    font-size: var(--oc-font-size-medium);
    min-height: $global-control-height;
  }

  &-l {
    @include oc-button-line-height(1.5);

    font-size: var(--oc-font-size-xlarge);
    min-height: 2rem;
  }

  &-primary {
    @include oc-button-color-role(var(--oc-role-primary), var(--oc-role-on-primary));
  }
  &-primary-container {
    @include oc-button-color-role(
      var(--oc-role-primary-container),
      var(--oc-role-on-primary-container)
    );
  }
  &-primary-fixed {
    @include oc-button-color-role(var(--oc-role-primary-fixed), var(--oc-role-on-primary-fixed));
  }
  &-secondary {
    @include oc-button-color-role(var(--oc-role-secondary), var(--oc-role-on-secondary));
  }
  &-secondary-container {
    @include oc-button-color-role(
      var(--oc-role-secondary-container),
      var(--oc-role-on-secondary-container)
    );
  }
  &-secondary-fixed {
    @include oc-button-color-role(
      var(--oc-role-secondary-fixed),
      var(--oc-role-on-secondary-fixed)
    );
  }
  &-tertiary {
    @include oc-button-color-role(var(--oc-role-tertiary), var(--oc-role-on-tertiary));
  }
  &-tertiary-container {
    @include oc-button-color-role(
      var(--oc-role-tertiary-container),
      var(--oc-role-on-tertiary-container)
    );
  }
  &-tertiary-fixed {
    @include oc-button-color-role(var(--oc-role-tertiary-fixed), var(--oc-role-on-tertiary-fixed));
  }
  &-surface {
    @include oc-button-color-role(var(--oc-role-surface), var(--oc-role-on-surface));
  }
  &-surface-container {
    @include oc-button-color-role(var(--oc-role-surface-container), var(--oc-role-on-surface));
  }
  &-chrome {
    @include oc-button-color-role(var(--oc-role-chrome), var(--oc-role-on-chrome));
  }

  &:hover:not(.no-hover, .oc-button-raw-inverse, .oc-button-raw, .active, .selected, [disabled]) {
    filter: brightness(85%);
  }

  &-outline:hover:not(.no-hover, [disabled]) {
    background-color: var(--oc-role-surface-container);
    filter: none !important;
  }

  &:disabled {
    cursor: default;
    opacity: 0.6;
  }

  &-group {
    display: inline-flex;
    flex-flow: row wrap;
    outline: 1px solid var(--oc-role-secondary);
    outline-offset: -1px;
    border-radius: 5px;

    .oc-button {
      border-radius: 0;
      outline: 0;

      &:first-of-type {
        border-radius: 5px 0 0 5px;
      }

      &:last-of-type {
        border-radius: 0 5px 5px 0;
      }
    }
  }
}
</style>
