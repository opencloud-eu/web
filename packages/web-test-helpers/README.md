# web-test-helpers

This packages provides utilities for unit testing within the OpenCloud app ecosystem.

## Installation

Depending on your package manager, run one of the following commands:

```
$ npm install @opencloud-eu/web-test-helpers --save-dev

$ pnpm add -D @opencloud-eu/web-test-helpers

$ yarn add @opencloud-eu/web-test-helpers --dev
```

Make sure that you have `vue`, `@vue/test-utils`, and `@opencloud-eu/web-pkg` as dev dependencies of your app since those are peer dependencies of this package.
