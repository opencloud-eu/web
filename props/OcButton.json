[
  {
    "name": "appearance",
    "required": false,
    "description": "The appearance of the button.",
    "default": "outline",
    "type": "AppearanceType"
  },
  {
    "name": "ariaLabel",
    "required": false,
    "description": "The aria label of the button. Needs to be present if the button doesn't have a visible label.",
    "default": "",
    "type": "string"
  },
  {
    "name": "colorRole",
    "required": false,
    "description": "Material design color role.",
    "default": "secondary",
    "type": "| 'primary'\n    | 'primaryContainer'\n    | 'primaryFixed'\n    | 'secondary'\n    | 'secondaryContainer'\n    | 'secondaryFixed'\n    | 'tertiary'\n    | 'tertiaryContainer'\n    | 'tertiaryFixed'\n    | 'surface'\n    | 'surfaceContainer'\n    | 'chrome'"
  },
  {
    "name": "disabled",
    "required": false,
    "description": "Determines if the button is disabled.",
    "default": "false",
    "type": "boolean"
  },
  {
    "name": "gapSize",
    "required": false,
    "description": "The gap size between content elements of the button.",
    "default": "medium",
    "type": "SizeType | 'none'"
  },
  {
    "name": "href",
    "required": false,
    "description": "The href if the `type` is set to `a'.",
    "default": "",
    "type": "string"
  },
  {
    "name": "justifyContent",
    "required": false,
    "description": "The alignment of the button content.",
    "default": "center",
    "type": "JustifyContentType"
  },
  {
    "name": "showSpinner",
    "required": false,
    "description": "Determines if a spinner should be shown inside the button.",
    "default": "false",
    "type": "boolean"
  },
  {
    "name": "size",
    "required": false,
    "description": "The size of the button.",
    "default": "medium",
    "type": "'small' | 'medium' | 'large'"
  },
  {
    "name": "submit",
    "required": false,
    "description": "The type of the button element. Only takes effect if the `type` is set to `button`.",
    "default": "button",
    "type": "'null' | 'button' | 'submit' | 'reset'"
  },
  {
    "name": "target",
    "required": false,
    "description": "The target of the button if the `type` is set to `a`.",
    "default": "",
    "type": "'_blank' | '_self' | '_parent' | '_top'"
  },
  {
    "name": "to",
    "required": false,
    "description": "The route location if the `type` is set to `router-link`.",
    "default": "",
    "type": "RouteLocationRaw"
  },
  {
    "name": "type",
    "required": false,
    "description": "The type of the button element.",
    "default": "button",
    "type": "'button' | 'a' | 'router-link'"
  },
  {
    "name": "noHover",
    "required": false,
    "description": "Determines if the button should have no hover effect.",
    "default": "false",
    "type": "boolean"
  }
]