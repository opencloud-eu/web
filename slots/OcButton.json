[
  {
    "name": "default",
    "description": "Button content.",
    "type": "() => unknown"
  }
]