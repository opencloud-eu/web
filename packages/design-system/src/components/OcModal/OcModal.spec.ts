import { defaultPlugins, mount, shallowMount } from '@opencloud-eu/web-test-helpers'
import Modal from './OcModal.vue'
import OcButton from './../OcButton/OcButton.vue'

const defaultProps = {
  title: 'Example title',
  message: 'Example message'
}

const inputProps = {
  title: 'Create new folder',
  hasInput: true,
  inputValue: 'New folder',
  inputLabel: 'Folder name'
}

describe('OcModal', () => {
  it('displays correct variation', () => {
    const wrapper = shallowMount(Modal, {
      props: {
        ...defaultProps,
        variation: 'danger'
      },
      global: { renderStubDefaultSlot: true, plugins: [...defaultPlugins()] }
    })

    expect(wrapper.findAll('.oc-modal-danger').length).toBe(1)
  })

  it('hides icon if not specified', () => {
    const wrapper = shallowMount(Modal, {
      global: { renderStubDefaultSlot: true, plugins: [...defaultPlugins()] },
      props: {
        ...defaultProps
      }
    })

    expect(wrapper.findAll('.oc-icon').length).toBe(0)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('overrides props message with slot', () => {
    const wrapper = shallowMount(Modal, {
      global: { renderStubDefaultSlot: true, plugins: [...defaultPlugins()] },
      props: {
        ...defaultProps
      },
      slots: {
        content: '<p>Slot message</p>'
      }
    })

    expect(wrapper.find('.oc-modal-body-message > p').text()).toMatch('Slot message')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(Modal, {
      global: { renderStubDefaultSlot: true, plugins: [...defaultPlugins()] },
      props: {
        ...defaultProps,
        icon: 'info'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('displays input', () => {
    const wrapper = shallowMount(Modal, {
      global: { renderStubDefaultSlot: true, plugins: [...defaultPlugins()] },
      props: inputProps
    })

    expect(wrapper.findAll('.oc-modal-body-input').length).toBe(1)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('displays loading state', async () => {
    const waitForSpinnerToShow = async () => {
      await wrapper.vm.$nextTick()
      return new Promise((resolve) => setTimeout(resolve, 1000))
    }

    const wrapper = mount(Modal, {
      global: {
        renderStubDefaultSlot: true,
        plugins: [...defaultPlugins()],
        stubs: {
          'focus-trap': true
        }
      },
      props: {
        ...defaultProps,
        isLoading: true
      }
    })

    const cancelButton = wrapper.find('.oc-modal-body-actions-cancel')
    const confirmButton = wrapper.find('.oc-modal-body-actions-confirm')

    expect(cancelButton.attributes('disabled')).toBeDefined()
    expect(confirmButton.attributes('disabled')).toBeDefined()
    expect(
      wrapper.findComponent<typeof OcButton>('.oc-modal-body-actions-confirm').props('showSpinner')
    ).toBeFalsy()
    expect(
      wrapper.findComponent<typeof OcButton>('.oc-modal-body-actions-confirm').props('appearance')
    ).toEqual('filled')

    await waitForSpinnerToShow()

    expect(
      wrapper.findComponent<typeof OcButton>('.oc-modal-body-actions-confirm').props('showSpinner')
    ).toBeTruthy()
    expect(
      wrapper.findComponent<typeof OcButton>('.oc-modal-body-actions-confirm').props('appearance')
    ).toEqual('outline')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
