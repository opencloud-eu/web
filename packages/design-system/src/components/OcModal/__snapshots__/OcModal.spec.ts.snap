// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OcModal > displays input 1`] = `
"<div class="oc-modal-background">
  <focus-trap-stub tabbableoptions="[object Object]" active="true" escapedeactivates="true" returnfocusondeactivate="true" allowoutsideclick="true" clickoutsidedeactivates="false" initialfocus="[Function]" delayinitialfocus="true" preventscroll="false" setreturnfocus="false">
    <div class="oc-modal" tabindex="0" role="dialog" aria-modal="true" aria-labelledby="oc-modal-title">
      <div class="oc-modal-title">
        <h2 id="oc-modal-title" class="oc-text-truncate">Create new folder</h2>
      </div>
      <div class="oc-modal-body">
        <!--v-if-->
        <!--v-if-->
        <oc-text-input-stub label="Folder name" id="oc-textinput-1" type="text" modelvalue="New folder" clearbuttonenabled="false" clearbuttonaccessiblelabel="" disabled="false" fixmessageline="true" readonly="false" passwordpolicy="[object Object]" class="oc-modal-body-input"></oc-text-input-stub>
      </div>
      <div class="oc-modal-body-actions oc-flex oc-flex-right">
        <div class="oc-modal-body-actions-grid">
          <oc-button-stub appearance="outline" colorrole="secondary" disabled="false" gapsize="medium" justifycontent="center" showspinner="false" size="medium" submit="button" type="button" nohover="false" class="oc-modal-body-actions-cancel">Cancel</oc-button-stub>
          <oc-button-stub appearance="filled" colorrole="secondary" disabled="false" gapsize="medium" justifycontent="center" showspinner="false" size="medium" submit="button" type="button" nohover="false" class="oc-modal-body-actions-confirm oc-ml-s">Confirm</oc-button-stub>
        </div>
      </div>
    </div>
  </focus-trap-stub>
</div>"
`;

exports[`OcModal > displays loading state 1`] = `
"<div class="oc-modal-background">
  <focus-trap-stub tabbableoptions="[object Object]" active="true" escapedeactivates="true" returnfocusondeactivate="true" allowoutsideclick="true" clickoutsidedeactivates="false" initialfocus="[Function]" delayinitialfocus="true" preventscroll="false" setreturnfocus="false">
    <div class="oc-modal" tabindex="0" role="dialog" aria-modal="true" aria-labelledby="oc-modal-title">
      <div class="oc-modal-title">
        <h2 id="oc-modal-title" class="oc-text-truncate">Example title</h2>
      </div>
      <div class="oc-modal-body">
        <p class="oc-modal-body-message oc-mt-rm oc-mb-rm">Example message</p>
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class="oc-modal-body-actions oc-flex oc-flex-right">
        <div class="oc-modal-body-actions-grid"><button type="button" disabled="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-outline oc-button-secondary-outline oc-modal-body-actions-cancel">
            <!--v-if-->
            <!-- @slot Content of the button --> Cancel
          </button> <button type="button" disabled="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-outline oc-button-secondary-outline oc-modal-body-actions-confirm oc-ml-s"><span class="oc-spinner oc-spinner-s spinner" aria-label="" tabindex="-1" role="img"></span> <!-- @slot Content of the button --> Confirm</button></div>
      </div>
    </div>
  </focus-trap-stub>
</div>"
`;

exports[`OcModal > matches snapshot 1`] = `
"<div class="oc-modal-background">
  <focus-trap-stub tabbableoptions="[object Object]" active="true" escapedeactivates="true" returnfocusondeactivate="true" allowoutsideclick="true" clickoutsidedeactivates="false" initialfocus="[Function]" delayinitialfocus="true" preventscroll="false" setreturnfocus="false">
    <div class="oc-modal" tabindex="0" role="dialog" aria-modal="true" aria-labelledby="oc-modal-title">
      <div class="oc-modal-title">
        <h2 id="oc-modal-title" class="oc-text-truncate">Example title</h2>
      </div>
      <div class="oc-modal-body">
        <p class="oc-modal-body-message oc-mt-rm oc-mb-rm">Example message</p>
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class="oc-modal-body-actions oc-flex oc-flex-right">
        <div class="oc-modal-body-actions-grid">
          <oc-button-stub appearance="outline" colorrole="secondary" disabled="false" gapsize="medium" justifycontent="center" showspinner="false" size="medium" submit="button" type="button" nohover="false" class="oc-modal-body-actions-cancel">Cancel</oc-button-stub>
          <oc-button-stub appearance="filled" colorrole="secondary" disabled="false" gapsize="medium" justifycontent="center" showspinner="false" size="medium" submit="button" type="button" nohover="false" class="oc-modal-body-actions-confirm oc-ml-s">Confirm</oc-button-stub>
        </div>
      </div>
    </div>
  </focus-trap-stub>
</div>"
`;

exports[`OcModal > overrides props message with slot 1`] = `
"<div class="oc-modal-background">
  <focus-trap-stub tabbableoptions="[object Object]" active="true" escapedeactivates="true" returnfocusondeactivate="true" allowoutsideclick="true" clickoutsidedeactivates="false" initialfocus="[Function]" delayinitialfocus="true" preventscroll="false" setreturnfocus="false">
    <div class="oc-modal" tabindex="0" role="dialog" aria-modal="true" aria-labelledby="oc-modal-title">
      <div class="oc-modal-title">
        <h2 id="oc-modal-title" class="oc-text-truncate">Example title</h2>
      </div>
      <div class="oc-modal-body">
        <div class="oc-modal-body-message">
          <p>Slot message</p>
        </div>
      </div>
      <div class="oc-modal-body-actions oc-flex oc-flex-right">
        <div class="oc-modal-body-actions-grid">
          <oc-button-stub appearance="outline" colorrole="secondary" disabled="false" gapsize="medium" justifycontent="center" showspinner="false" size="medium" submit="button" type="button" nohover="false" class="oc-modal-body-actions-cancel">Cancel</oc-button-stub>
          <oc-button-stub appearance="filled" colorrole="secondary" disabled="false" gapsize="medium" justifycontent="center" showspinner="false" size="medium" submit="button" type="button" nohover="false" class="oc-modal-body-actions-confirm oc-ml-s">Confirm</oc-button-stub>
        </div>
      </div>
    </div>
  </focus-trap-stub>
</div>"
`;
